hw1-app/src/app/hwork1/search/search.component.ts

```diff
@@ -10,17 +12,24 @@ export class SearchComponent implements OnInit {
   @Output() searchStringOnChange = new EventEmitter<string>();
 
   public searchString: string;
+  public subject = new Subject<string>();
 
   constructor() {
   }
 
   ngOnInit(): void {
     this.searchString = '';
+    this.subject.pipe(filter(text => text.length >= 3),
+      debounceTime(500))
```

Comment:
It will be good to add distinctUntilChanged() operator to the end of pipe chain to prevent subscription firing with equal values

Reply:
Thanks. Good idea